### PR TITLE
Move hardcoded exceptions rules to more specific rules and unify exception handling

### PR DESCRIPTION
**Pull request by bbondy**
Related issue: [brave/brave-browser#3475](https://github.com/brave/brave-browser/issues/3475)'
Related PR: https://github.com/brave/brave-core/pull/1770

This is the first part of a series of work that will be done within the next few weeks. 
It unifies exception handling in one place.

In a later PR, we will be adding metadata to filter rules, and exposing options to control those rules.
It will be opt-out options at first, and then a later iteration will try going to opt-in with UX to notify the user that things are broken.

See this issue for a description of the follow up work that will be done:
https://github.com/brave/brave-browser/issues/3475

This is specifically for:
- Facebook login buttons
- Facebook embed posts
- Twitter embeds

Note that this does not change existing functionality, it just moves a hardcoded c++ list to a more unified place where we can iterate on to add options to exclude them.

For a test plan, see the linked issue above.

### CODE REVIEW

**Comment by bbondy** [comment]
Pls don't merge if approved.

Note that this blocks more strictly than disconnect but I didn't see an issue caused by it.  
We'd have to add an extra allow exception I think to make it parity.

File: brave-unbreak.txt [hunk 1]
```diff
@@ -88,3 +87,27 @@
...
+@@||connect.facebook.net/*/sdk.js$script
+@@||facebook.com/connect/
+@@||www.facebook.com/connect
+@@||staticxx.facebook.com/connect/
+@@||graph.facebook.com/
+! Block fbcdn.net everywhere but allow Facebook embeds
+||fbcdn.net$third-party,domain=~facebook.com
+@@||staticxx.facebook.com/
+@@||xx.fbcdn.net/
+@@||www.facebook.com/plugins/post.php?href=
```
Comment (fmarier) [inline brave-unbreak.txt]: There are [a few more URLs](https://github.com/brave/brave-browser/issues/3241#issuecomment-465825247) for Facebook embeds:

- `www.facebook.com/plugins/video.php`
- `www.facebook.com/plugins/comment_embed.php`
- `www.facebook.com/plugins/page.php`

If you want to test them all at once, you an use the [test page I created last week](https://fmarier.github.io/brave-testing/social-widgets.html).

**Comment by fmarier** [comment]
Should we also move [these exceptions](https://github.com/brave/tracking-protection/blame/132ac934be9dd0113a4bb4d041f7602a72c60662/data/disconnect.json#L9348-L9375) to `brave-unbreak.txt` (if they're even needed at all)?

File: brave-unbreak.txt [hunk 2]
```diff
@@ -88,3 +87,36 @@
...
+@@||facebook.com/connect/
+@@||www.facebook.com/connect
+@@||staticxx.facebook.com/connect/
+@@||graph.facebook.com/
+! Block fbcdn.net everywhere but allow Facebook embeds
+||fbcdn.net$third-party,domain=~facebook.com
+@@||staticxx.facebook.com/
+@@||xx.fbcdn.net/
+@@||www.facebook.com/*/plugin
+@@||www.facebook.com/plugins/
```
Comment (fmarier) [inline brave-unbreak.txt]: If we unblock all of the plugins, then we'll let through the Like/Share buttons that @tomlowenthal wanted us to block in brave/brave-browser#3241:

- `www.facebook.com/plugins/like.php`
- `www.facebook.com/plugins/share_button.php`

**Comment by fmarier** [comment]
Looping in @tomlowenthal because we may not want all of the widgets from <https://fmarier.github.io/brave-testing/social-widgets.html> to be allowed by default.

I think we definitely want Facebook videos to work for example, but maybe the LinkedIn Share/Follow social buttons aren't valuable enough to users warrant the tracking by Microsoft.

**Comment by bbondy** [comment]
@fmarier @tomlowenthal I don't want to have that discussion here since this task is scoped to be about moving things from hardcoded lists to a unified exception list that we can build off of.  I'll remove the LinkedIn one and do it in another PR that we can only merge once the option to control it exists.

**Comment by bbondy** [comment]
Mainly I suggest we avoid such discussions because if we just follow the 3 step plan I outlined in the section "Summary of the plan" from https://github.com/brave/brave-browser/issues/3475 then we can completely skip the discussions which have no clear way to decide what's best globally.

**Comment by bbondy** [comment]
Talked to @fmarier on Slack and he's ok merging this since it retains the existing functionality.  Going ahead.

**Pull request merged**